---
author: "[Nima Hejazi](https://nimahejazi.org), Amanda Mok, Courtney Schiffman"
title: "Discovering Cancer Signatures via Non-Negative Matrix Factorization"
institution: "University of California, Berkeley"
date: "`r Sys.Date()`"
bibliography: refs.bib
output: binb::metropolis
---

```{r,setup, include=FALSE}
knitr::opts_chunk$set(cache=TRUE)
```

# Introduction (Nima)

## Overview and Motivations

* ...

* ...

* ...

## Overview of Matrix Factorization

* Matrix factorization as unsupervised learning

* What can we learn about objects by matrix factorization?

* A general formulation of matrix factorization

* Various forms of matrix factorization: NMF, PCA, VQ

* Applications of matrix factorization: images, text

* Biological applications of matrix factorization

# Non-Negative Matrix Factorization (Nima)

## What is Matrix Factorization?

* Suppose we have a _data matrix_ $V$ of dimension $n \times m$, each column of
  which is an $n$-vector of observations of a given variable.

* A factorization of $V$ produces two matrices $\{W, H\}$ that approximately
  capture the information present in $V$.

* From linear algebra, we have $V_{ij} \approx (W H)_{ij} = \sum_{a = 1}^r
  W_{ia} H_{aj}$.

* The dimensionality of the induced matrix factors is reduced wrt $V$ -- that
  is, let $W$ be $n \times r$ and $H$ be $r \times m$.

* This can be viewed as a form of data compression when the rank $r$ is small in
  comparison to $n$ and $m$.
  * In particular, $r$ is often chosen such that $(n + m)r \leq nm$.

## What is Matrix Factorization?

* With the general factorization $V_{ij} \approx \sum_{a = 1}^r W_{ia} H_{aj}$,
  $W$ and $H$ each pick up different important aspects of $V$.

* When $V$ is a $n \times m$ matrix of images of faces, where each row
  corresponds to a pixel and each column an image:
  * the $r$ columns of $W$ may be thought of as basis images,
  * and each of the $j$ columns of $H$ is termed an encoding (coefficients to be
    applied to basis images).

* Various forms of matrix factorization place different types of constraints on
  the manner in which $W$ and $H$ are generated.

## Vector Quantization (VQ)

* A form of matrix factorization in which the matrix factor $H$ is constrained
  such that each column has a single entry equal to unity, with all other
  entries being zero.

* Since this is a constraint on the _encoding_ columns, this results in each
  column of $W$ representing some distortion of the target image.

* Equivalently, each column of $V$ is approximated by a single basis (column of
  $W$).

* In terms of image learning, this results in the VQ decomposition learning
  _prototypical_ faces.

## VQ: Prototypical Faces

![](./figs/vq_faces.png){height=90%}

## Principal Components Analysis (PCA)

* ...

* ...

## PCA: _Eigenfaces_

![](./figs/pca_faces.png){height=90%}

## PCA in Biology

* Obligatory example: John Novembre's European populations

## What is NMF?

* ...

* ...

* ...

## Who cares about non-negativity?

* ...

* ...

* ...

## What does non-negativity buy us?

* ...

* ...

* ...


## NMF: Parts of Faces

![](./figs/nmf_faces.png){height=90%}

## Some fun with NMF

* ...

* ...

* ...

## NMF in biology

* example from Bioconductor?

* pretty plot goes here

## NMF in cancer biology

* So, we've now established that NMF finds _parts_ of the input matrix through
  the non-negativity constraint it imposes on the matrix factors.

* This has important applications for exploring cancer biology; namely, applying
  NMF could help us detect _parts of tumors_.

* Interpretation is challenging: does this mean we're detecting subclonal
  populations?

* There's a whole lot more to come.

# A bit of biology (Amanda)

## What is cancer?

* Complex tissues with multiple cell types and interactions

* Characterized by unchecked somatic cell proliferation

* Normal cells acquire hallmark traits that enable them to become tumorigenic ^[@hanahan2011hallmarks]

## Hallmarks of Cancer

![Hallmarks of Cancer](./figs/cancer_hallmarks.jpg)

## Cancer is a genetic disease

* Germline mutations: inherited from parents

    * Mutations in tumor suppressor genes or oncogenes can predispose someone to develop cancer

* Somatic mutations: acquired over time in somatic cells

    * Endogenous: DNA damage as a result of metabolic byproducts

    * Exogenous: DNA damage as a result of mutagenic exposure

* Epigenetic modifications: no change to DNA sequence

    * DNA methylation

    * Histone modification

    * MicroRNA gene silencing

## Somatic mutations

* Rearrangements

* Copy number changes

* Indels

* Base substitutions

    * 6 types of substitutions (C>G, C>T, C>A, G>T, G>A, T>A)

    * 4 types of 5' base nucleotide

    * 4 types of 3' base nucleotide

    * Transcriptional strand

## Clonal evolution in cancer

# Applying NMF to a biological challenge

## @alexandrov2013deciphering characterize mutational processess as a blind source separation problem

Mutational catalogs "are the cumulative result of all the somatic mutational
mechanisms ...that have been operative during the cellular lineage starting from
the fertilized egg...to the cancer cell."

![](./figs/biol_challenge.jpg){height=45%}

## NMF is a natural method for handling the BSS problem

![](./figs/biol_challenge.jpg){height=60%}

* Non-negative matrix entries.

* Want to learn the parts (mutational signatures of mutational processes) that
  add to the whole (mutational catalog).

## What are the basis vectors and encodings in the context of mutational processes?

![](./figs/Screen Shot 2018-09-30 at 7.30.25 PM.png)
$$M \approx P\times E$$

$M$, $K$ mutation types by $G$ genomes

$P$, $K$ mutation types by $N$ mutation signatures

$E$, $N$ mutation signatures by $G$ genomes


## What are the basis vectors and encodings in the context of mutational processes?

![](./figs/Screen Shot 2018-09-30 at 7.44.14 PM.png){width=80%}

![](./figs/Screen Shot 2018-10-01 at 8.25.26 AM.png){width=30%}



## The parts that make up the whole in mutational processes

![](./figs/part_of_whole.jpg)

## Method for deciphering signatures of mutational processes

1. Input matrix $M$ of dimension $K$ (mutation types) by $G$ (genomes).

2. Remove rare mutations (< 1%).

3. Monte Carlo bootstrap resampling.


## Method for deciphering signatures of mutational processes

4. Apply the multiplicative update algorithm until convergence.
* Repeat steps 3 and 4 $I$ times, each time storing $P$ and $E$.
* Typical values $I=400-500$


![](./figs/Screen Shot 2018-09-28 at 8.35.41 AM.png){width=35%}

![](./figs/Screen Shot 2018-09-28 at 8.35.52 AM.png){width=30%}



## Method for deciphering signatures of mutational processes

5. Cluster the signatures (columns of $P$ matrix) from the $I$ iterations into $N$ clusters, one signature per cluster for each of the $I$ matrices.
  * This automatically clusters the exposures.
  * Use cosine similarity for clustering.

![](./figs/clusters.jpeg){width=40%}


## Method for deciphering signatures of mutational processes

6. Create the iteration averaged centroid matrix, $\overline{\rm P}$, by averaging the signatures within each cluster.

7. Evaluate the reproducibility of the signatures by calculating the average silhouette width over the $N$ clusters.

![](./figs/silhouette.png){width=50%}


## Method for deciphering signatures of mutational processes

8. Evaluate the accuracy of the approximation of $M$ by calculating the Frobenius reconstruction errors.

![](./figs/Screen Shot 2018-09-28 at 8.35.41 AM.png){width=40%}

9. Repeat steps 1-8 for different values of $N=1,\dots,min(K,G)-1$.

## Method for deciphering signatures of mutational processes

10. Choose an $N$ corresponding to highly reproducible mutational signatures and low reconstruction error.

![](./figs/Screen Shot 2018-09-30 at 8.55.52 PM.png){height=40%}


## The method is affected by the number of genomes, uniqueness of signatures, and number of mutations

![](./figs/simulations.jpg)

## The method recovers 10 signatures in a simulated cancer genome dataset


![](./figs/100genome.jpg)

## Findings (Amanda)

* ...

* ...

* ...

# We've talked enough (Amanda)

## Discussion

* ...

* ...

* ...

## References

